### ISSUE TITLE
Unique page <title>s for each webpage

### ISSUE THREAD

**Issue by MaddyDaigle** [GCTC-NTGC/gc-digital-talent#4657]
## Background

The individual page `<title>`s are not unique and are all “GC Digital Talent”. Each page should have a unique title so that users can navigate through multiple open tabs.

## Recommendation
Follow this convention for each page `<title>`:

- H1 | GC Digital Talent
- e.g. If the user was on the profile main page = “My Profile | GC Digital Talent”

## Documentation
@MaddyDaigle will be using the [site list](https://go.talent.c4nada.ca/sitemap) to identify all the page titles.

## A11y
Here’s some WCAG documentation on this issue:
Failure of success criterion 2.4.2: Page titled
https://www.w3.org/WAI/WCAG21/Understanding/page-titled.html
Failures
The following are common mistakes that are considered failures of this Success Criterion by the WCAG Working Group.
[F25: Failure of Success Criterion 2.4.2 due to the title of a Web page not identifying the contents](https://www.w3.org/WAI/WCAG21/Techniques/failures/F25)

## Acceptance criteria

- [ ] List of page titles are added to the [site list](https://go.talent.c4nada.ca/sitemap)

**Comment by esizer** [comment]
I think as part of this, we should create an `<SEO />` component. This will allow us to use open graph meta data as well from simpler input in the future.

## Example

```tsx

const SEO = ({title, description}: SEOProps) => {
  const intl = useIntl();
  const location = useLocation();
  const siteUrl = SITE_URL_CONSTANT; //comes from env?
  const defaultTitle = intl.formatMessage({
    defaultMessage: "GC Digital Talent",
    ...
  });
  const defaultDescription = '...';

  const seo ={
    title: title ? `${title} | ${defaultTitle}` : defaultTitle,
    description: description || defaultDescription,
    url: `${siteUrl}${location.pathname}`
  }

  return (
    <Helmet>
        <title>{seo.title}</title>
        <meta name="description" content={seo.description} />
    </Helmet>
  )
}

/**
 * Usage
 */
import SEO from "@common/component/SEO/SEO";

const Page = () => {

  return (
    <>
      <SEO title="Profile" />
      {/* ... */}
    </>
  )
}
```

**Comment by MaddyDaigle** [comment]
All the unique page titles have been added to the [site list](https://go.talent.c4nada.ca/sitemap).  @gobyrne @esizer 

A couple determinations before dev can start:

1. Should we provide context in the page title?

**Example:** When the user is editing a specific pool, should the page title say:
Edit Pool | Admin - GC Digital Talent
or
Edit [Pool name] | Admin - GC Digital Talent

Context is generally better, but I worry that the page titles will get too long.  Thoughts?

2. What should we call the admin application?
Right now I have it as "Admin - GC Digital Talent".  Any other (shorter) suggestions that still provide context?

**Comment by esizer** [comment]
> 1. Should we provide context in the page title?

Maybe we could provide the context in the description? I do know that [ideally titles are fewer than 60 characters in length](https://moz.com/learn/seo/title-tag#:~:text=While%20Google%20does,in%20the%20SERPs) so we should strive for that if possible and I do know some of our titles for things like pools are longer than 60 characters in some cases.

**Comment by gobyrne** [comment]
I updated the spreadsheet, should be good to go now.
